### PR TITLE
front: NavBar: display license attributions

### LINKED ISSUE #3255: Properly credit front-end libs
### What happened?

Many licenses require being mentioned / credited in the final software. We aren't doing it (because we forgot), and we should fix that at some point

### What did you expect to happen?

Credit libraries

### OSRD version

e6976222

### PR DESCRIPTION
**Pull request by Yohh**
close #3255

### CODE REVIEW

File: front/public/locales/fr/translation.json [hunk 1]
```diff
@@ -305,7 +305,8 @@
   },
   "NavBar": {
     "disconnect": "Déconnexion",
-    "informations": "Informations"
+    "informations": "Informations",
+    "attributions": "Attributions"
```
Comment (multun) [inline front/public/locales/fr/translation.json]: can you merge both sections into a single "About" button? that's the way it's usually done

**Review by multun — CHANGES_REQUESTED** [review]

File: front/public/locales/fr/translation.json (hunk 1 above)
Comment (Yohh) [reply front/public/locales/fr/translation.json]: done

File: front/public/locales/fr/translation.json [hunk 2]
```diff
@@ -304,8 +304,10 @@
     "traincompo": "Matériel"
   },
   "NavBar": {
+    "about": "A propos",
     "disconnect": "Déconnexion",
-    "informations": "Informations"
+    "informations": "Informations",
```
Comment (clarani) [inline front/public/locales/fr/translation.json]:
```suggestion
```
Not used anymore, to remove

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        <h3 className="mr-1">{item.package.data.name}</h3>{' '}
+        <span className="informations-version">{`(${item.package.data.version})`}</span>
+      </div>
+      <span className="pl-4 mb-4">
+        {item.package.data.copyright ||
+          item.package.data.publisher ||
+          `${item.package.data.name} authors`}
+      </span>
+    </div>
+  ));
```
Comment (clarani) [inline front/src/common/ReleaseInformations/LicenseAttributions.tsx]:
```suggestion
const attributions = Object.values(licenses).map(({ name, version, copyright, publisher }) => (
    <div className="d-flex flex-column align-items-start" key={name}>
      <div className="d-flex">
        <h3 className="mr-1">{name}</h3>{' '}
        <span className="informations-version">{`(${version})`}</span>
      </div>
      <span className="pl-4 mb-4">{copyright || publisher || `${name} authors`}</span>
    </div>
  ));
```
This is enough :) you don't need to make things complicated haha

Comment (clarani) [reply front/src/common/ReleaseInformations/LicenseAttributions.tsx]: You can also remove the License type which is not used anymore !

File: front/src/common/ReleaseInformations/consts.ts [hunk 4]
```diff
@@ -0,0 +1,717 @@
...
+  };
+}
+
+// to generate the json file in order to get licenses informations
+// we have to use license-checker-rseidelsohn and yui-lint
+// then this bash command while located in /front:
+//
+// license-checker-rseidelsohn --json --production --direct --excludePackagesStartingWith "@types" --excludePrivatePackages --customPath src/common/ReleaseInformations/json/licenseCustomFormat.json > src/common/ReleaseInformations/json/licenses.json
+//
+// then copy/paste the result in the const below
```
Comment (clarani) [inline front/src/common/ReleaseInformations/consts.ts]: Great !! Can you create a script in package.json to do this (for instance, generate-license) ?
You can also add a small text about this in the README of front :)

File: front/src/common/ReleaseInformations/consts.ts [hunk 5]
```diff
@@ -0,0 +1,717 @@
...
+  name: string;
+  version: string;
+}
+
+export interface License {
+  package: {
+    title: string;
+    data: Data;
+  };
+}
```
Comment (clarani) [inline front/src/common/ReleaseInformations/consts.ts]:
```suggestion
```
Not useful :)

**Review by clarani — CHANGES_REQUESTED** [review]
Nice job ! I left some comments

File: front/public/locales/fr/translation.json (hunk 2 above)
Comment (Yohh) [reply front/public/locales/fr/translation.json]: done

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx (hunk 3 above)
Comment (Yohh) [reply front/src/common/ReleaseInformations/LicenseAttributions.tsx]: done

File: front/src/common/ReleaseInformations/consts.ts (hunk 5 above)
Comment (Yohh) [reply front/src/common/ReleaseInformations/consts.ts]: done

File: front/src/common/ReleaseInformations/consts.ts (hunk 4 above)
Comment (Yohh) [reply front/src/common/ReleaseInformations/consts.ts]: done

**Review by clarani — APPROVED** [review]
LGTM !! :partying_face:

**Review by multun — APPROVED** [review]

**Pull request merged**